ev3dev/helper.py

```diff
@@ -146,6 +147,171 @@ class MediumMotor(ev3dev.auto.MediumMotor, MotorMixin):
+            motor.command = kwargs['command']
+            log.info("%s: %s command %s" % (self, motor, kwargs['command']))
+
+    def run_forever(self, **kwargs):
+        kwargs['command'] = ev3dev.auto.LargeMotor.COMMAND_RUN_FOREVER
+        self._run_command(**kwargs)
+
+    def run_to_abs_pos(self, **kwargs):
+        kwargs['command'] = ev3dev.auto.LargeMotor.COMMAND_RUN_TO_ABS_POS
+        self._run_command(**kwargs)
+
+    def run_to_rel_pos(self, **kwargs):
+        kwargs['command'] = ev3dev.auto.LargeMotor.COMMAND_RUN_TO_REL_POS
+        self._run_command(**kwargs)
+
+    def run_timed(self, **kwargs):
+        kwargs['command'] = ev3dev.auto.LargeMotor.COMMAND_RUN_TIMED
+        self._run_command(**kwargs)
+
+    def run_direct(self, **kwargs):
+        kwargs['command'] = ev3dev.auto.LargeMotor.COMMAND_RUN_DIRECT
+        self._run_command(**kwargs)
+
+    def reset(self, motors=None):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            motor.reset()
+
+    def stop(self, motors=None):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            motor.stop()
+
+    def _is_state(self, motors, state):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            if state not in motor.state:
+                return False
+
+        return True
+
+    @property
+    def is_running(self, motors=None):
+        return self._is_state(motors, ev3dev.auto.LargeMotor.STATE_RUNNING)
+
+    @property
+    def is_ramping(self, motors=None):
+        return self._is_state(motors, ev3dev.auto.LargeMotor.STATE_RAMPING)
+
+    @property
+    def is_holding(self, motors=None):
+        return self._is_state(motors, ev3dev.auto.LargeMotor.STATE_HOLDING)
+
+    @property
+    def is_overloaded(self, motors=None):
+        return self._is_state(motors, ev3dev.auto.LargeMotor.STATE_OVERLOADED)
+
+    @property
+    def is_stalled(self):
+        return self._is_state(motors, ev3dev.auto.LargeMotor.STATE_STALLED)
+
+    def wait(self, cond, timeout=None, motors=None):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            motor.wait(cond, timeout)
+
+    def wait_until_not_moving(self, timeout=None, motors=None):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            motor.wait_until_not_moving(timeout)
+
+    def wait_until(self, s, timeout=None, motors=None):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            motor.wait_until(s, timeout)
+
+    def wait_while(self, s, timeout=None, motors=None):
+        motors = motors if motors is not None else self.motors.values()
+
+        for motor in motors:
+            motor.wait_while(s, timeout)
+
+
+class LargeMotorPair(MotorSet):
```

Comment:
I think we should figure out exactly what our goal is. In my mind, the most common scenario will be a pair of large motors which you want to "steer": even if internally we're using a generic `MotorSet`, I'd like to see if we can make the "pair" classes more specialized so they legitimately feel easy to use and work "out of the box".

Reply:
yeah....to me part of it depends on what the user input is to control the steering.  If it is the IR remote we have that implemented in the RemoteControlledTank class in helper.py...we could pull some of that code into a Mixin class that could be inherited here in LargeMotorPair.  I also did a web interface for a tank, that could be pulled in too but loading the www modules makes the program take a while to load on an ev3. What other user input methods might we see for a MotorPair?

Reply:
The scenarios I had in mind were:
- "Run the left motor at 50% speed forward and the right motor at 10% speed backward"
- "Run the two motors in a proportion of 3:1" (ratio, differential, etc.)

The second is a special case of the first. Note that the units are nonspecific and just an illustration. The idea is that this class is what's used whenever you have a driving robot... if you want to go straight, turn, spin, etc., you use this.